p2p/server.py

```diff
@@ -257,9 +257,18 @@ def _make_peer_pool(self, discovery: DiscoveryProtocol) -> PeerPool:
         if self.peer_pool.is_full:
             peer.disconnect(DisconnectReason.too_many_peers)
         else:
-            # We use self.wait() here as a workaround for
-            # https://github.com/ethereum/py-evm/issues/670.
-            await self.wait(self.do_handshake(peer))
+            total_peers = len(self.peer_pool.connected_nodes)
+            inbound_peers = len(
+                [peer for peer in self.peer_pool.connected_nodes.values() if peer.inbound])
+            if total_peers > 1 and inbound_peers / total_peers > 0.75:
+                # make sure to have at least 1/4 outbound connections
+                peer.disconnect(DisconnectReason.useless_peer)
```

Comment:
Maybe double check what `DisconnectReason` geth and parity use for that case. `useless_peer` doesn't strike me as right but the protocol doesn't specify something that matches really well. Maybe `too_many_peers` would make more sense?